SmurfTestingScripts/README.md

```diff
@@ -2,34 +2,54 @@
+Test is successful when SmurfEvaluator is successful. The script considers the exit code of the application run.
 
+### Sequence diagram
+
+```mermaid
+sequenceDiagram
+    User->>Script: Execute smurf_compare_scenarios.py
+    Script->>validate_json: Validate scenario JSON
+    validate_json-->>Script: JSON Valid
+    Script->>setup_env: Setup test environment
+    setup_env->>Script: Execute setup commands
+    loop For each scenario
+        Script->>run_scenarios: Run scenario
+        run_scenarios->>run_scenarios: Start the tested executable
+        run_scenarios->>run_scenarios: Start actions
+        run_scenarios->>run_scenarios: Compare scenario output
+        run_scenarios->>Script: Tidy up between scenarios
+    end
+    Script->>cleanup: Cleanup after all scenarios
+    cleanup->>Script: Execute cleanup commands
+    Script-->>User: Provide test results
+```
 
 ## Usage
-Scenario.json file contains commands that will be run before, after and in between each application run.
-Application runs contains name of test, timeout in seconds, and program arguments.
+The scenario json file contains commands that will run before, in between each application run, and after all runs are completed.
+Application runs contain the name of a test, timeout in seconds, program arguments and commands to be run in parallel.
 
-All paths used in scenario.json are relative to this file.
+All paths used in the scenario json are relative to that file.
 
 **Created directories:** 
-* etalons: created when -c option is used, files with raw .log files
+* etalons: created when -c option is used; files with raw .log files
 * output: raw .log output of test-runs
 * aggregated_output: aggregated output of test-runs
-* evaluator_output: SmurfEvaluator compare output containing differences between etalon and output
+* evaluator_output: SmurfEvaluator output containing the comparison between an etalon and an output
 
 ## Run
 ```
-python3 CompareScenarios.py --scenario <path> --executable <path> --evaluator <path> [--output <path> --create-etalons]
+python3 CompareScenarios.py --scenario <path> --evaluator <path> [--executable <path> --output <path> --create-etalons --env <path>]
 ```  
-first run must  have option --create-etalons! Etalons aren't created automatically when not found,
-because they demand human-approve.
+The first run must be run with the --create-etalons flag! Etalons aren't created automatically when not found because they require human approval. The --executable argument is not needed if it is provided in the scenario file. The json file provided for the --env argument is used to parametrize the scenario file (strings used for key names will be replaced by their values in the scenario file if they are contained within brackets of a special STATE_SMURF_ENV[] string).
```

Comment:
Are there checks in cross - when the executable is specified in .json and by --executable what is desired behavior?

Reply:
the executable provided by the command argument has priority; mentioned in readme now